bitswap/client/client.go

```diff
@@ -97,6 +98,27 @@ func WithoutDuplicatedBlockStats() Option {
 	}
 }
 
+// WithContentRouting enables content.Routing lookups to find providers for
+// the content to be retrieved using Bitswap.
+func WithContentRouting(router routing.Routing) Option {
```

Comment:
We could have a `ProviderFinder` interface here instead of Routing. I think users may not necessarily figure out that a ProviderFinder is a subset of their already existing content routing though, so it is more self-documenting this way.

Reply:
I find the coupling of these unfortunate and non-Golike as it can end up resulting in wrapper layers and such. It's a bit like how the libp2p host is passed around everywhere even when only a fraction of its functionality is needed. Would just documenting this function and calling out that a `routing.ContentRouting` (https://github.com/libp2p/go-libp2p/blob/dc97b73e4d2970142e366e047a34853a5f54fd8c/core/routing/routing.go#L26) also suffices.

Note: even routing.Routing is way bigger than what we need. If breaking apart the interface in go-libp2p/core/routing would be helpful I'd be that'd be reasonable to the go-libp2p maintainers.

Reply:
Let's do that... I will open a PR on libp2p to split it and we can swap later.

Reply:
https://github.com/libp2p/go-libp2p/pull/3048